In [96]:
import sys
import os

os.chdir("../")

os.getcwd()

'c:\\'

In [97]:
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn import tree

from source.data_preprocess import DataPreprocessing
from source.model_builder import ModelBuilder

In [98]:
preprocessor = DataPreprocessing()

data = preprocessor.load_data('C:\\Users\kterri3\Documents\Git Repositories\meters\meters\data\Meter_A.txt')

         0         1         2         3          4          5         6   \
0  0.841499  1.009367  0.993816  8.469805  10.278727  10.037759  8.501365   
1  0.842250  1.006584  0.996605  7.531891   9.139924   8.951618  7.612213   
2  0.840723  1.011647  0.998152  6.641699   7.975464   7.857692  6.593117   
3  0.841119  1.017807  0.996812  5.687524   6.824334   6.689885  5.615428   
4  0.840358  1.016534  0.996221  5.660385   6.829560   6.675628  5.623977   

         7          8          9   ...         27         28         29  \
0  8.581726  10.247763  10.058822  ...  32.451173  34.568685  33.082683   
1  7.623325   9.106345   8.945142  ...  32.428385  34.441732  33.081055   
2  6.681572   7.964596   7.814698  ...  32.428385  34.275715  33.113605   
3  5.763315   6.801051   6.686639  ...  32.485350  34.080403  33.170573   
4  5.736818   6.813453   6.672377  ...  32.503255  34.122720  33.164062   

          30         31         32         33         34         35  36  
0  36.722005

In [99]:
train_validation, test = train_test_split(data, test_size = 0.2, random_state=12)

train, validation = train_test_split(train_validation, test_size = 0.2, random_state=99)

In [100]:
scaler = StandardScaler().fit(train[:, :-1])

train_scaled = scaler.transform(train[:, :-1])

In [101]:
model_builder_ANN = ModelBuilder()

ANN_model = model_builder_ANN.dt(X_train=train[:, :-1], X_test=validation[:, :-1], y_train=train[:, -1], y_test = validation[:, -1])

model_builder_ANN.accuracy

0.8571428571428571

Default Parameters: 

0.85714 

Trying new hyperparameter combinations....

Test #1: "hidden_layer_sizes"=50

0.78571

Test #2: "hidden_layer_sizes"=100

0.92857

Test #3: "learning_rate_init=2"

0.92857

Test #4: "learning_rate_init=5"

0.92857

Test #5: "hidden_layer_sizes"=100 and "learning_rate_init=5"

0.92857

It appears that with each combination the highest accuracy achieved is 92.86%.